cluster/addons/iso-client/README.md

```diff
@@ -0,0 +1,95 @@
+### Cgroup-cpuset-cpus isolator.
+
+This is an example implementation of extended pod-level resource isolation as specified [here](https://github.com/kubernetes/features/issues/246).
+Cgroup-cpuset-cpus isolator aims to provide a way for an operator to specify which PODs should be run with exclusive [cpusets](http://man7.org/linux/man-pages/man7/cpuset.7.html).
+
+### Building 
+
+In order to use this feature, one has to build a modified kubelet which resides in Intel's fork of Kubernetes [here](https://github.com/intelsdi-x/kubernetes/tree/ext-iso).
+
+First issue:
+
+```
+hack/update-generated-runtime-dockerized.sh
+```
+
+to generate lifecycle protobuff files, and then compile kubelet with:
+
+
+```
+build/run.sh make KUBE_FASTBUILD=true ARCH=amd64
+```
+
+a kubelet binary can be found here:
+
+```
+_output/dockerized/bin/linux/amd64/kubelet
+```
+
+Every node which is supposed to be able to use cgroup-cpuset-cpus isolator has to use this kubelet, additionaly to enable extended pod-level resource isolation kubelet has to be run with two additional flags:
+
+```
+--cgroups-per-qos=true --cgroup-root=/
+```
+
+For extended pod-level resources to be easily plugable, kubelet spawns another service called EventDispatcher and custome isolators are registering to it in order to receive IsolationRequests. That is why we suggest to run custom isolators as DaemonSets and orchestrate their placements with node labels and NodeSelectors.
+
+To create a docker image with cgroup-cpuset-cpus isolator execute this script (resulting docker images name is passed as first argument):
+
+```
```

Comment:
Use snippet decorators to bring some helpful colors ;) For shell files use `sh` for json use `json`. So your snippet should start with \`\`\`sh

Example:
```
#!/usr/bin/env bash

while true; do
    uname -a
done
```

vs

```sh
#!/usr/bin/env bash

while true; do
    uname -a
done
```